keras/utils/generic_utils.py

```diff
@@ -461,6 +461,22 @@ def to_list(x):
     return [x]
 
 
+def first_or_list(x):
```

Comment:
It is not clear from the name "first_or_list" what the function does.

A list or set with a single element is a singleton. Maybe a better name would be `unpack_singleton`.

Reply:
Thanks for the name! I was looking for one that would be meaningful.